staging/kos/pkg/util/parse/network/subnet/summary.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subnet
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	MinVNI uint32 = 1
+	MaxVNI uint32 = 2097151
+)
+
+// Summary is a simplified representation of a KOS API Subnet object. It stores
+// the information KOS control plane components need to know to carry out their
+// tasks.
+type Summary struct {
+	NamespacedName types.NamespacedName
+	VNI            uint32
+
+	// BaseU is the lowest IPv4 address of the subnet as a uint32.
+	BaseU uint32
+
+	// LastU is the highest IPv4 address of the subnet as a uint32.
+	LastU uint32
+}
+
+func NewSummary(subnet interface{}) (*Summary, Errors) {
```

Comment:
There is a lot of mechanism here to accomplish very little.  I think there is a much simpler way to get the job done, and would prefer it.  See https://github.com/MikeSpreitzer/kube-examples/commit/3b8220fb88b7c48deef68d5ea60ed89c068b0094

Reply:
You're right, I was rather dissatisfied with the outcome, thanks for handling this. I cherry-picked your commit and tweaked a few things in 02883e7.